### ISSUE TITLE
Question: Regarding trouble logging into my Humble account via the App

### ISSUE THREAD

**Issue by SpraxDev** [Gustash/lutris_installers#18]
Hi!

I am seeing two similar errors and was hoping you (or somebody) could help me find the issue:

When I start the app through Lutris, it shows me that it got an `undefined` path:
![image](https://github.com/Gustash/lutris_installers/assets/7331598/5b32c6f5-1447-4cde-836c-7f7046200409)

If I try to click on one of the two login buttons, I get presented with another, seemingly path related error:
![image](https://github.com/Gustash/lutris_installers/assets/7331598/5e79bffd-08ad-49bb-80d6-ae1e4204c282)
`Ungültiger Name.` is German and basically translates to `Invalid name.`

Sadly no file paths or anything are presented to me but they *shouldn't* contain any non-ASCII characters.

---

Not sure if the app is actually bugging or the system/wine/... setup is not fully compatible with my linux installation. I *think* I saw somewhat similar error messages a couple months ago, but I sadly don't remember where.

**Comment by Gustash** [comment]
Hey. Maybe a recent update changed how login is handled. I'll try to debug this, but it might be hard since I'm no longer subscribed to Humble Choice. 

If it lets me login at least, I might be able to find a fix.

EDIT:

So I was able to login even without being subscribed. I also tried re-installing Humble Choice and login worked as before.

The only difference was that there seems to be a bug in Lutris where variable substitution isn't working as intended, so the executable path was pointing to `/start-humble.cmd` instead of `$GAMEDIR/start-humble.cmd`. But since you were able to start the launcher, that doesn't seem to be the issue here.

From the application code, you seem to be getting the first error when it's trying to figure out the install path from the Windows Registry, but I'm not sure why it's failing in your case. Maybe your account has some German characters that Humble can't handle? If so, that's an issue with the application itself and there's not much I can do...

**Comment by SpraxDev** [comment]
Thanks for checking! Do you think/know if there is anything I can do to debug or identify the issue? I'm usually just hitting play on Steam and am not used to Wine etc. 😄 

What I found in the registry looks fine to me:
![image](https://github.com/Gustash/lutris_installers/assets/7331598/b8a5f5c4-1cf4-4d32-81d3-6d5122a955a3)
(`HKEY_USERS\S-1-5-21-[...]\Software\2f793df2-2969-529d-b0c0-7960ed40d70e`)

---

Otherwise I'll just have to try and contact the support and see if they can do anything about that

**Comment by Gustash** [comment]
Yeah, that seems right... Have you tried re-installing?

**Comment by SpraxDev** [comment]
Multiple times sadly :( Using your yaml from this repo and the script on lutris (didn't check if they are the same).
I don't know any of the variables that actually affect this, so it's hard to change stuff and see if it helps 😄 

I'll close the issue then and maybe contact Humble's support. If they actually are able to help, I'll reply to the closed issue.
Thanks for your help so far!

**Comment by SpraxDev** [comment]
Sorry @Gustash but another question (you may just tell me to ask in a more public forum 😄):

I installed the Humble app via Steam (added the installer as a game) and can successfully launch it. My problem now is that I do not return from Humble's authorization page to the launcher. I see the `humble://` URI it is trying to point me to but I have no idea how to get it to that app :/

Do you know if I can add some arguments when launching the app that contains the code from that URI or something?

**Comment by Gustash** [comment]
You can check what this installer is doing. But basically that is handled by a `.desktop` file that tells the system how to handle `humble://` URIs.

You also need a script that "fixes" the URI, because for some reason the format the website returns is different than what the app itself expects.

You should be able to just get the contents of all these files in the yml file on this repo :)

**Comment by SpraxDev** [comment]
I just coundln't get it to work. I tried manually adding the *'patched'* URI as a start parameter but the App just doesn't show any reaction :/

I uninstalled my Steam install of the launcher and gave your Lutris script a last chance before I just have another reason to cancel my subscription but it just worked? No weird JS errors and the protocol/URI handler just worked instantly.

Thanks for all your efforts, it *now works on my machine* too 😄
